frontend/states/reducer.js

```diff
@@ -0,0 +1,13 @@
+import { actions } from "./index";
+
+export const reducer = (state, action) => {
```

Comment:
Using reducers seems overkill to me. The toast is only going to display a singe string that doesn't depend on previous states. Using a `useState`/`useMemo` like for the `AuthContext` would remove quite a bit of boilerplate.